src/ui/public/agg_types/agg_type.js

```diff
@@ -140,6 +140,18 @@ export function AggTypesAggTypeProvider(Private) {
      */
     this.decorateAggConfig = config.decorateAggConfig || null;
 
+    /**
+     * A function that will be called after the main request has been made
+     * and should return an updated response
+     * @param aggConfigs - agg config array used to produce main request
+     * @param aggConfig - AggConfig that requested the post flight request
+     * @param searchSourceAggs - SearchSource aggregation configuration
+     * @param resp - Response to the main request
+     * @param nestedSearchSource - the new SearchSource that will be used to make post flight request
+     * @return {Promise}
+     */
+    this.postFlightRequest = config.postFlightRequest || null;
```

Comment:
since we're keeping this `postFlightRequest`-approach, imo we should make it a first-class member of the `agg_type` object and make it as simple to use as possible form a client perspective.

To that end, I'd suggest following changes:

1) Implement by default with the identity function. E.g.

` this.postFlightRequest = config.postFlightRequest || function(resp){return resp}`;

2) simplify the method-contract

I don't think it;s necessary for clients to have to implement (or know how to implement) a searchSource

(1) and (2) should make the implementation in courier more simple, and should communicate better that this isn't optional. The courier code would then look something like
```
 }).then(async resp => {
       for (agg in aggs){
         resp =  await agg.type.postFlightRequest(resp, vis.aggs, vis.agg, searchSource);         
      }
      return resp;
    });
```